### PR TITLE
fix(yarn): improve yarn proxy configuration

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

- Add experimental flag to skip Yarn proxy setting
- Unset each variable before setting

## Context

#27983

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/npm/post-update/yarn.ts [hunk 1]
```diff
@@ -210,15 +210,19 @@ export async function generateLockFile(
...
-        `yarn config set --home httpProxy ${quote(process.env.HTTP_PROXY)}`,
-      );
-    }
-    if (process.env.HTTPS_PROXY && !isYarn1) {
-      commands.push(
-        `yarn config set --home httpsProxy ${quote(process.env.HTTPS_PROXY)}`,
-      );
+    if (!process.env.RENOVATE_X_YARN_IGNORE_PROXY) {
+      if (process.env.HTTP_PROXY && !isYarn1) {
+        commands.push('yarn config unset --home httpProxy');
```
Comment (viceice) [inline lib/modules/manager/npm/post-update/yarn.ts:215]: what does the unset improve? 🤔

Comment (rarkins) [reply lib/modules/manager/npm/post-update/yarn.ts:215]: Unset returns "nothing is set"

Comment (rarkins) [reply lib/modules/manager/npm/post-update/yarn.ts:215]: Actually should add an || true here

Comment (viceice) [reply lib/modules/manager/npm/post-update/yarn.ts:215]: an `|| true` will maybe fail on windows?

Comment (rarkins) [reply lib/modules/manager/npm/post-update/yarn.ts:215]: Do we have any safe way to ignore the result of such a command and continue?

Comment (viceice) [reply lib/modules/manager/npm/post-update/yarn.ts:215]: it seems we do this on other places too 😕 
https://github.com/renovatebot/renovate/blob/c93cd9616fcf8509cd7e9efd3577a395d916d45a/lib/modules/manager/composer/artifacts.ts#L172
https://github.com/renovatebot/renovate/blob/c93cd9616fcf8509cd7e9efd3577a395d916d45a/lib/modules/manager/npm/post-update/yarn.ts#L159
https://github.com/renovatebot/renovate/blob/c93cd9616fcf8509cd7e9efd3577a395d916d45a/lib/util/exec/containerbase.ts#L342

Comment (rarkins) [reply lib/modules/manager/npm/post-update/yarn.ts:215]: Let's leave as-is in this PR

**Review by viceice — APPROVED** [review]

**Pull request merged**